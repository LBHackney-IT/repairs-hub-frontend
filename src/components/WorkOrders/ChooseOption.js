import PropTypes from 'prop-types'
import Radios from '../Form/Radios/Radios'
import BackButton from '../Layout/BackButton/BackButton'
import { useForm } from 'react-hook-form'
import { PrimarySubmitButton } from '../Form'
import { useRouter } from 'next/router'

const ChooseOption = ({ reference }) => {
  const router = useRouter()

  const { handleSubmit, register, errors } = useForm({
    mode: 'onChange',
  })

  const onSubmitForm = (e) => {
    if (e.options == 'Close job') {
      router.push({
        pathname: 'close-job',
        query: { id: reference },
      })
    } else if (e.options == 'Update') {
      router.push({
        pathname: 'update-job',
        query: { id: reference },
      })
    }
  }

  return (
    <>
      <BackButton />
      <h1 className="govuk-heading-l">Update work order: {reference}</h1>
      <form role="form" onSubmit={handleSubmit(onSubmitForm)}>
        <Radios
          label="Select process"
          name="options"
          options={['Update', 'Close job']}
          register={register({
            required: 'Please select a process',
          })}
          error={errors && errors.options}
        />
        <PrimarySubmitButton label="Next" />
      </form>
    </>
  )
}
ChooseOption.propTypes = {
  reference: PropTypes.string.isRequired,
}

export default ChooseOption
