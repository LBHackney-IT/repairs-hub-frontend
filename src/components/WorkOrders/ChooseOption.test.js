import { render } from '@testing-library/react'
import ChooseOption from './ChooseOption'

describe('ChooseOption component', () => {
  const props = {
    reference: '00012345',
  }

  it('should render properly', () => {
    const { asFragment } = render(<ChooseOption reference={props.reference} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
