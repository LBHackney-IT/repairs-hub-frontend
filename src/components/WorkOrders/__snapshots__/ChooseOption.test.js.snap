// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseOption component should render properly 1`] = `
<DocumentFragment>
  <a
    class="govuk-back-link"
    role="button"
  >
    Back
  </a>
  <h1
    class="govuk-heading-l"
  >
    Update work order: 00012345
  </h1>
  <form
    role="form"
  >
    <div
      class="govuk-form-group"
    >
      <label
        class="govuk-label govuk-label--m"
        for="options"
      >
        Select process 
      </label>
      <br />
      <div
        class="govuk-radios"
      >
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="options_Update"
            name="options"
            type="radio"
            value="Update"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="options_Update"
          >
            Update
          </label>
        </div>
        <div
          class="govuk-radios__item"
        >
          <input
            class="govuk-radios__input"
            id="options_Close job"
            name="options"
            type="radio"
            value="Close job"
          />
          <label
            class="govuk-label govuk-radios__label"
            for="options_Close job"
          >
            Close job
          </label>
        </div>
      </div>
    </div>
    <div
      class="govuk-form-group"
    >
      <button
        class="govuk-button"
        data-module="govuk-button"
        type="submit"
      >
        Next
      </button>
    </div>
  </form>
</DocumentFragment>
`;
